### PR TITLE
feat: adds CurrencyInput component

### PR DESCRIPTION
**Pull request by alansouzati**
#### What does this PR do?
adds a new component called CurrencyInput

#### Where should the reviewer start?
storybook examples and components/CurrencyInput
#### What testing has been done on this PR?
manual and automated
#### How should this be manually tested?
open storybook examples and tinker with the 3 examples added
#### Do the grommet docs need to be updated?
yes
#### Should this PR be mentioned in the release notes?
yes
#### Is this change backwards compatible or is it a breaking change?
backwards compatible

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -51,6 +51,7 @@
   "dependencies": {
     "grommet-icons": "^4.5.0",
     "hoist-non-react-statics": "^3.2.0",
+    "locale-currency": "^0.0.2",
```
Comment (ShimiSun) [inline package.json:54]: image

Doesn't look too bad, yet another package to maintain.

**Comment by ericsoderberghp** [comment]
- I like having the ability for intelligent numeric separators. I think we should have this capability for numbers in general, not just for currency.
- It would be nice to have the onChange provide both the text and numeric values, so the caller doesn’t also have to parse the text to extract the number.
- I’d be more inclined to explore adding this to MaskedInput, to see how that looks.
- we’re a bit reluctant to add more input components, as that incurs additional maintenance costs (minor)

**Review by ShimiSun — COMMENTED** [review]
Thank you for this PR, it drove a good conversation! 
In addition to eric's comments above, I'm adding some comments regarding the deployment/functionality

1. I think there is a small bug when numbers are entered after the dot, as you add a number it keeps increasing the value in a decimal value instead of typing the number.
2. On the "Simple" example, it is not clear that the user should skip the typing of the dollar sign.
3. I think we must have an example of each input component inside of a Form.

**Comment by alansouzati** [comment]
No worries, I will have this component on our end. Thanks for considering it.

File: package.json (hunk 1 above)
Comment (alansouzati) [reply package.json:54]: this package is a simple map of locale to currency, if you are concerned about the extra dependency I think there are maps available out there

```
{
   "pt-BR": "BRL",
  "pt": "EUR",
  "en-UK": "EUR",
  "en-US": "USD",
  "en": "USD"
}
```

**Pull request closed without merging**